how-to/security/openssh-server.md

```diff
@@ -65,35 +67,57 @@ Many other configuration directives for `sshd` are available to change the serve
 
 ## SSH keys
 
-SSH allows authentication between two hosts without the need of a password. SSH key authentication uses a **private key** and a **public key**.
+SSH allows authentication between two hosts without the need of a password, using cryptographic keys instead.
+{term}`SSH-key` authentication uses a **private key** and a **public key**.
 
-To generate the keys, run the following command:
+To generate a key pair, run the following command:
 
 ```bash
-ssh-keygen -t rsa
+ssh-keygen -t ed25519
 ```
 
-This will generate the keys using the **RSA Algorithm**. At the time of this writing, the generated keys will have 3072 bits. You can modify the number of bits by using the `-b` option. For example, to generate keys with 4096 bits, you can use:
+If needed, you can generate the key using the **RSA Algorithm** by using `-t rsa -b 4096`.
+
+During creation, you will be prompted for a key password, which you would enter to use that key (every time, or instead using an [ssh agent](#ssh-agent).
+
+By default, the public key is saved in the file `~/.ssh/id_ed25519.pub`, while `~/.ssh/id_ed25519` is the private key.
+To allow login for a user via key, append the content of `id_ed25519.pub` to `target_machine:~/.ssh/authorized_keys` by running:
 
 ```bash
-ssh-keygen -t rsa -b 4096
+ssh-copy-id username@remotehost
 ```
 
-During the process you will be prompted for a password. Simply hit <kbd>Enter</kbd> when prompted to create the key.
-
-By default, the public key is saved in the file `~/.ssh/id_rsa.pub`, while `~/.ssh/id_rsa` is the private key. Now copy the `id_rsa.pub` file to the remote host and append it to `~/.ssh/authorized_keys` by running:
+Finally, double check the permissions on the `authorized_keys` file -- only the authenticated user must have write permissions.
+If the permissions are not correct then change them by:
 
 ```bash
-ssh-copy-id username@remotehost
+chmod go-w .ssh/authorized_keys
 ```
 
-Finally, double check the permissions on the `authorized_keys` file -- only the authenticated user should have read and write permissions. If the permissions are not correct then change them by:
+You should now be able to SSH to the `target_machine` without being prompted for a password.
+
+To troubleshoot this, have a look at the `target_machine`'s journal:
 
 ```bash
-chmod 600 .ssh/authorized_keys
+sudo journalctl -eu sshd.service
+```
+
+
+## Connection multiplexing
+
+To reduce connection times, multiple SSH sessions to the same `target_machine` can work over one {term}`TCP` connection.
```

Comment:
suggestion: "To reduce initial connection times", or "To reduce connection setup times".